core/src/com/unciv/logic/automation/unit/CityLocationTileRanker.kt

```diff
@@ -0,0 +1,135 @@
+package com.unciv.logic.automation.unit
+
+import com.unciv.logic.automation.Automation
+import com.unciv.logic.civilization.Civilization
+import com.unciv.logic.civilization.diplomacy.DiplomacyFlags
+import com.unciv.logic.map.mapunit.MapUnit
+import com.unciv.logic.map.tile.Tile
+import com.unciv.models.ruleset.tile.ResourceType
+import com.unciv.models.ruleset.tile.TileResource
+import kotlin.math.max
+import kotlin.math.min
+
+class CityLocationTileRanker {
+    companion object {
+        fun getBestTilesToFoundCity(unit: MapUnit): List<Pair<Tile, Float>> {
+            val modConstants = unit.civ.gameInfo.ruleset.modOptions.constants
+            val tilesNearCities = sequence {
+                for (city in unit.civ.gameInfo.getCities()) {
+                    val center = city.getCenterTile()
+                    if (unit.civ.knows(city.civ) &&
+                            // If the CITY OWNER knows that the UNIT OWNER agreed not to settle near them
+                            city.civ.getDiplomacyManager(unit.civ)
+                                .hasFlag(DiplomacyFlags.AgreedToNotSettleNearUs)
+                    ) {
+                        yieldAll(
+                            center.getTilesInDistance(6)
+                                .filter { it.isExplored(unit.civ) || unit.civ.isAI() })
+                        continue
+                    }
+                    yieldAll(
+                        center.getTilesInDistance(modConstants.minimalCityDistance)
+                            .filter { it.isExplored(unit.civ) || unit.civ.isAI() }
+                            .filter { it.getContinent() == center.getContinent() }
+                    )
+                    yieldAll(
+                        center.getTilesInDistance(modConstants.minimalCityDistanceOnDifferentContinents)
+                            .filter { it.isExplored(unit.civ) || unit.civ.isAI() }
+                            .filter { it.getContinent() != center.getContinent() }
+                    )
+                }
+            }.toSet()
+
+            // This is to improve performance - instead of ranking each tile in the area up to 19 times, do it once.
+            val nearbyTileRankings = getNearbyTileRankings(unit.getTile(), unit.civ)
+
+            val distanceFromHome = if (unit.civ.cities.isEmpty()) 0
+            else unit.civ.cities.minOf { it.getCenterTile().aerialDistanceTo(unit.getTile()) }
+            val range = max(
+                1,
+                min(5, 8 - distanceFromHome)
+            ) // Restrict vision when far from home to avoid death marches
```

Comment:
how about 
```suggestion
            val range = (8 - distanceFromHome).coerceIn(1,5) // Restrict vision when far from home to avoid death marches
```

Reply:
Done